include/mp/proxy-types.h

```diff
@@ -421,24 +408,15 @@ struct ClientParam
 
     struct ReadResults : IterateFieldsHelper<ReadResults, sizeof...(Types)>
     {
-        template <typename... Args>
-        void handleField(Args&&... args)
-        {
-            callRead<0>(std::forward<Args>(args)...);
-        }
-
-        template <int I, typename... Args>
-        auto callRead(Args&&... args) -> std::enable_if_t<(I < sizeof...(Types))>
+        template <typename Results, typename... Params>
+        void handleField(ClientInvokeContext& invoke_context, Results& results, TypeList<Params...>)
         {
-            callRead<I + 1>(std::forward<Args>(args)..., std::get<I>(m_client_param->m_values));
-        }
+            auto const fun = [&]<typename... Values>(Values&&... values) {
```

Comment:
In commit "replace custom tuple unpacking code with `std::apply`" (36898dbfc72c9748f54bf79f18d1ad6485f48920)

Note: `Values` type here is currently unused, so lambda declaration could be simplified to just take an `auto&& values` parameter. However, keeping `Values` is nice even though it is unused because it makes ReadParams and BuildResults code more consistent. Also it might be useful to add perfect forwarding for `ReadDestUpdate(std::forward<Values>(values))` in the future. That will require other changes to this code however, so better to leave alone for now.

In general there are a **lot** more simplifications that can be made here now that this code no longer needs to work with c++11. Would be good to follow up in a separate PR.